sklearn/tests/test_grid_search.py

```diff
@@ -315,9 +316,10 @@ def test_grid_search_sparse():
     C = cv.best_estimator_.C
 
     X_ = sp.csr_matrix(X_)
+    y_ = sp.csr_matrix(y_.reshape((200, 1)))
```

Comment:
This will now be treated as a multilabel problem, rather than binary (even if the results will be the same for width 1). So this is a little confusing, particularly, as now you must (according to the warning now output by the test, I hope) use 'f1_micro' or 'f1_macro' or 'f1_weighted' rather than 'f1' as the scorer.

You could use `make_multilabel_classification` to produce a more sensible sparse (in the sense of often having zero-valued entries) `X` and `y`.